File: Striver Series/2FindTheSecondLargestAndSecondSmallestElementInAnArray.cpp
//Problem Statement: Given an array, find the second smallest and second largest element in the array. Print ‘-1’ in the event that either of them doesn’t exist.
#include<iostream>
using namespace std;
class Solution{
    public:
    pair<int,int> solv(int arr[],int size){
        for(int j=0;j<size-1;j++){
            for(int i=0;i<size-1-j;i++){
            if(arr[i]>arr[i+1]){
                int store=arr[i];
                arr[i]=arr[i+1];
                arr[i+1]=store;
            }
        }
        }
        return {arr[1],arr[size-1]};
    }
};
int main(){
    int arr[]={3,4,5,2,3};
    int size=sizeof(arr)/sizeof(arr[0]);
    Solution sl;
    pair<int,int>result=sl.solv(arr,size);
    cout<<result.first<<endl;
    cout<<result.second;
}

File: python/2FindTheSecondLargestAndSecondSmallestElementInAnArray.py
'''
Problem Statement: Given an array, find the second smallest and second largest element in the array. Print ‘-1’ in the event that either of them doesn’t exist.

Example 1:
Input: [1,2,4,7,7,5]
Output: Second Smallest : 2
	    Second Largest : 5
Explanation: The elements are as follows 1,2,3,5,7,7 and hence second largest of these is 5 and second smallest is 2
'''
class Solution:
    def Solv(self,list):
        largest=list[0]
        smallest=list[0]
        for i in list:
            if i>largest:
                largest=i
            if i<smallest:
                smallest=i
        print(largest,'\t',smallest)
        sec_largest=float('-inf')
        sec_smallest=float('inf')
        for i in list:
            if i>sec_largest and i<largest:
                sec_largest=i
            if i<sec_smallest and i>smallest:
                sec_smallest=i
        print(sec_largest,'\t',sec_smallest)
list=[1,2,3,7,7,5]
sl=Solution()
sl.Solv(list)
